subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/JavaCompile.java

```diff
@@ -375,9 +375,21 @@ DefaultJavaCompileSpec createSpec() {
 
     private void configureCompatibilityOptions(DefaultJavaCompileSpec spec) {
         if (javaCompiler.isPresent()) {
-            final JavaInstallationMetadata toolchain = javaCompiler.get().getMetadata();
-            spec.setTargetCompatibility(toolchain.getLanguageVersion().asString());
-            spec.setSourceCompatibility(toolchain.getLanguageVersion().asString());
+            if (compileOptions.getRelease().isPresent()) {
```

Comment:
This would fail with a java 6/7 toolchain, no? But yeah, if that is the target, a user should not specific the release flag anyway

Reply:
Yes, that the principle I took when re-introducing the `release` flag and did not find an easy way to figure out early with which Java version we were going to run.